processor/src/main/java/org/mapstruct/ap/internal/util/NativeTypes.java

```diff
@@ -233,6 +221,185 @@ private boolean floatHasBecomeZero() {
         }
     }
 
+   private static class BooleanAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            if ( !( "true".equals( s ) || "false".equals( s ) ) ) {
+                throw new IllegalArgumentException("only 'true' or 'false' are supported");
+            }
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return boolean.class;
+        }
+    }
+
+    private static class CharAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            if ( !(s.length() == 3 && s.startsWith( "'" ) && s.endsWith( "'" )) ) {
+                throw new NumberFormatException( "invalid character literal" );
+            }
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return char.class;
+        }
+    }
+
+    private static class ByteAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            NumberRepresentation br = new NumberRepresentation( s, true, false, false ) {
+
+                @Override
+                void parse(String val, int radix) {
+                    Byte.parseByte( val, radix );
+                }
+            };
+            br.validate();
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return int.class;
+        }
+    }
+
+    private static class DoubleAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            NumberRepresentation br = new NumberRepresentation( s, false, false, false ) {
+
+                @Override
+                void parse(String val, int radix) {
+                    Double d = Double.parseDouble( radix == 16 ? "0x" + val : val );
+                    if ( doubleHasBecomeZero( d  ) ) {
+                        throw new NumberFormatException( "floating point number too small" );
+                    }
+                    if ( d.isInfinite() ) {
+                        throw new NumberFormatException( "infinitive is not allowed" );
```

Comment:
This is odd. On one hand, the user needs to specify a number as a string, not `Double.POSITIVE_INFINITY`, so this is a logical restriction to have. But I can definitely imagine cases where "infinity" is a reasonable default/constant value to use (say, as a sentinel value).

Perhaps both `0` and infinite should be treated as warnings and not errors?

Reply:
Since we have full freedom, we could introduce a magic string: e.g. 'INFINITIVE'. I'm not in favour of having unintentional infinitive. That's why I decided to have a mapping error.

Reply:
I agree that unintentional infinity should be detected, along with unintentional `0`. It's easy to imagine that causing a lot of pain, especially when it's hidden in generated code. I just worry it's too restrictive in some cases; plus it is not be backward compatible with current behaviour.

Magic strings is a good idea. With good errors ("did you mean to use ZERO?"), you don't really need to go to the docs each time you want a zero or infinite value. These also aren't backward compatible, but good error messages make for a simple upgrade path at least.

Reply:
I think we should handel zero the normal way 😄 . However, we might want to borrow some keywords from the `Double` and `Float` definition. MapStruct would assign them automatically. I need to think though if I solve it in this PR or in a separate one. We can refer to those in the javadoc. We could refer to those in (the proper) error messages as you suggest.

Natural candidates would be (from `Float` and `Double` definition):  `POSITIVE_INFINITY`,`NEGATIVE_INFINITY`,`NaN`,`MAX_VALUE `, `MIN_NORMAL`, `MIN_VALUE`

Reply:
Re-read... `NEGATIVE_MAX_VALUE` is less ambiguous.

Reply:
Sounds good to me.